### PR TITLE
switch to using TensorFlow as backend for recent versions of Keras

### PR DESCRIPTION
**Pull request by boegel**
Now that we can properly build `TensorFlow` from source (cfr. https://github.com/easybuilders/easybuild-easyblocks/pull/1287), we should also use `TensorFlow` as backend for Keras (which is the default), since it performs a lot better than `Theano` (which is pretty much end of life).

requires ~~#5819~~ (`TensorFlow`)

cc @JensTimmerman

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2441.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/dad0d3fb966edb15e675e5e7ba3c0943 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2035.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/ceccf32166a3ff395ea35e705381bdcf for a full test report.

**Comment by JensTimmerman** [comment]
On a test application (faceswap) I saw a 5x speed improvement in iteration speed (with similar loss) between a TensorFlow built from source vs theano (built without blas and lapack it warned me) backend.
So yay for tensorflow.

**Comment by boegel** [comment]
See issue reported by @JensTimmerman at https://github.com/keras-team/keras/issues/9381 .

Since this is a very clear error, which possible only occurs in same contexts, I don't think that's a blocker for this particular PR...
